### DISCUSSION TITLE
How to "connect" existing Cloudinary images (instead of uploading)

### DISCUSSION THREAD

**Discussion by jesstelford** [keystonejs/keystone#7934] [Questions]
The `CloudinaryImage` type currently requires a file upload during update/create. However, there are cases such as migrating from an existing platform, or populating a test database where being able to connect to existing images already uploaded to cloudinary would be desired.

This may need to be a breaking change as the current input object for update/create on a `CloudinaryImage` field is expected to always be an `Upload` type:

```graphql
type PostCreateInput {
  title: String
  image: Upload
}
```

I propose we move the `Upload` type down a level and introduce a new option `connect`:

```graphql
type PostCreateInput {
  title: String
  image: CloudinaryImageCreateInput
}

input CloudinaryImageCreateInput {
  upload: Upload
  connect: CloudinaryImageConnectInput
}

input ConnectCloudinaryImageInput {
  cloudinaryId: ID!
  # ... Any other unique identifying info we need
}
```

The implementation then would simply query Cloudinary for the unique info provided rather than first uploading anything.

If both `upload` & `connect` are set, an error is raised.

**Comment by VinayaSathyanarayana** [comment]
Agree with @jesstelford

**Comment by dijs** [comment]
Agreed. And what happened to the `select` feature?

Since, once again, my images are already in cloudinary.

**Comment by jesstelford** [comment]
It would also be nice to support adding images to cloudinary via URI:

```graphql
type PostCreateInput {
  title: String
  image: CloudinaryImageCreateInput
}

input CloudinaryImageCreateInput {
  upload: Upload
  connect: CloudinaryImageConnectInput
  fromUri: String
}
```

**Comment by ghost** [comment]
Any workaround?

**Comment by ogheneovo12** [comment]
any workaround or fixes
